core/config.js

```diff
@@ -0,0 +1,58 @@
+Nuxeo = Nuxeo || {};
+Nuxeo.UI = Nuxeo.UI || {};
+Nuxeo.UI.config = Nuxeo.UI.config || {};
+const { config } = Nuxeo.UI;
+
+Object.assign(config, {
+  /**
+   * Returns a property for a given `path`. If it is not defined, `fallback` is returned instead.
+   * If `fallback` is defined, the method will try to convert the property value to the same data type of `fallback`.
+   */
+  get(path, fallback) {
+    let val = path.split('.').reduce((a, b) => a && a[b], this);
+    if (fallback !== undefined && typeof val !== typeof fallback) {
+      let type;
+      switch (typeof fallback) {
+        case 'boolean':
+          type = Boolean;
+          break;
+        case 'number':
+          type = Number;
+          break;
+        case 'string':
+          type = String;
+          break;
+        case 'bigint':
+          // eslint-disable-next-line no-undef
+          type = BigInt;
+          break;
+        case 'undefined':
+        case 'symbol':
+        default:
+          break;
+      }
+      val = (val !== undefined && type && type(val)) || val;
```

Comment:
perhaps we should skip the type conversion straight away when `val` is undefined

Reply:
Agree